plat/imx/common/imx_sip_handler.c

```diff
@@ -168,14 +169,37 @@ int imx_src_handler(uint32_t smc_fid,
 	case IMX_SIP_SRC_IS_SECONDARY_BOOT:
 		val = mmio_read_32(IMX_SRC_BASE + SRC_GPR10_OFFSET);
 		return !!(val & SRC_GPR10_PERSIST_SECONDARY_BOOT);
+#endif /* defined(PLAT_imx8mm) || defined(PLAT_imx8mq) */
+#if defined(PLAT_imx8mm)
+	case IMX_SIP_SRC_START_M_CORE:
+		val = mmio_read_32(IMX_SRC_BASE + SRC_M4RCR_OFFSET);
+		val &= ~SRC_SW_M4C_NON_SCLR_RST;
+		val |= SRC_ENABLE_M4;
+		mmio_write_32(IMX_SRC_BASE + SRC_M4RCR_OFFSET, val);
+		break;
+	case IMX_SIP_SRC_IS_M_CORE_STARTED:
+		val = mmio_read_32(IMX_SRC_BASE + SRC_M4RCR_OFFSET);
+		return !(val & SRC_SW_M4C_NON_SCLR_RST);
+#endif /* defined(PLAT_imx8mm) */
+#if defined(PLAT_imx8mn) || defined(PLAT_imx8mp)
+	case IMX_SIP_SRC_START_M_CORE:
+		val = mmio_read_32(IMX_SRC_BASE + SRC_M7RCR_OFFSET);
```

Comment:
for imx8mn and imxmp, here set/get GPR

	case IMX_SIP_SRC_M4_START:
		mmio_clrbits_32(IMX_IOMUX_GPR_BASE + 0x58, 0x1);
		break;
	case IMX_SIP_SRC_M4_STARTED:
		val = mmio_read_32(IMX_IOMUX_GPR_BASE + 0x58);
		return !(val & 0x1);

Reply:
Not sure which line of code you refer with this block... I was not able to find the mentioned block anywhere in the source tree.

Any reference to the source code would be appreciated here.